crates/ruma-events/src/call.rs

```diff
@@ -44,6 +47,41 @@ impl SessionDescription {
     }
 }
 
+/// Metadata about a VoIP stream.
+#[derive(Clone, Debug, Serialize, Deserialize)]
+#[cfg_attr(not(feature = "unstable-exhaustive-types"), non_exhaustive)]
+pub struct StreamMetadata {
+    /// The purpose of the stream.
+    pub purpose: StreamPurpose,
+}
+
+impl StreamMetadata {
+    /// Creates a new `StreamMetadata` with the given purpose.
+    pub fn new(purpose: StreamPurpose) -> Self {
+        Self { purpose }
+    }
+}
+
+/// The purpose of a VoIP stream.
+#[doc = include_str!(concat!(env!("CARGO_MANIFEST_DIR"), "/src/doc/string_enum.md"))]
+#[derive(Clone, PartialEq, Eq, StringEnum)]
+#[ruma_enum(rename_all = "m.snake_case")]
+#[non_exhaustive]
+pub enum StreamPurpose {
+    /// `m.usermedia`.
+    ///
+    /// A stream that contains the webcam and/or microphone tracks.
+    Usermedia,
```

Comment:
Is not really a word... Do we have per-variant name overrides so this can be `UserMedia` but use the non-eord-separated string form?

Reply:
No, that's why I used this :stuck_out_tongue: 

But it would be easy to add. It's just a shame they can't stick to a single convention when naming fields…

Reply:
Added the new rule

Reply:
I meant whether we support `rename()` on an individual variant instead of a `rename_all` rule that skips underscores, but I guess this works too.